### ISSUE TITLE
Duplicate check-in when reentering app left on checkin response screen

### ISSUE THREAD

**Issue by GoogleCodeExporter** [iamakshayv/foursquared#111]
```
What steps will reproduce the problem?
1. Check-in to a venue and wait for the response screen to be fully loaded 
(the one mentioning your total # of checkins and current mayor)
2. Leave the application by pressing home.
3. Re-open Foursquare.

What is the expected output? What do you see instead?

Ideally I'd see the screen just as I left it (or even forgetting it and going 
back to the main venue page would be fine). Instead it tries to reload the 
response screen and checks in a second time in the process.

What version of the product are you using? On what operating system?

DRC79, although I've seen it on other builds as well.

Please provide any additional information below.

```

Original issue reported on code.google.com by `bli...@gmail.com` on 27 Sep 2009 at 1:46

**Comment by GoogleCodeExporter** [comment]
```
blipus,

Running the same build as you, I don't see this as repeatable. Can you repeat 
it? How 
did you get to the venue in the first place, shortcut, main screen, or search?
```

Original comment by `jlapenna` on 27 Sep 2009 at 2:54

**Comment by GoogleCodeExporter** [comment]
```
So it turns out you can't just leave the app and re-open it. You need to do 
enough 
stuff outside of the app to get the process (or at least the rendering of that 
screen) pushed out of memory. On my G1 it was enough to just use the browser, 
maps, 
and read some email. I'm guessing it's a distinction between that checkin code 
getting called from OnCreate, OnStart, or OnResume (in 
http://developer.android.com/reference/android/app/Activity.html#ActivityLifecyc
le) 
but I'm really not sure.

FWIW, I got to the venue by searching for it.
```

Original comment by `bli...@gmail.com` on 29 Sep 2009 at 2:15

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 3230c4dc95.
```

Original comment by `jlapenna` on 5 Oct 2009 at 4:24
- Changed state: **Fixed**
